### PR TITLE
Add utility class that apply preventive remedial actions on a network object

### PR DESCRIPTION
**Pull request by murgeyseb**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the new behavior (if this is a feature change)?**
Add a utility class that can be used to apply found preventive remedial actions on an input network.

### CODE REVIEW

File: data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * Apply preventive remedial actions saved in CRAC result extension on current working variant of given network.
+     *
+     * @param network Network on which remedial actions should be applied
+     * @param crac CRAC that should contain result extension
+     * @param cracVariantId CRAC variant to get active remedial actions from
+     */
+    public static void applyPreventiveRemedialActions(Network network, Crac crac, String cracVariantId) {
+        String preventiveStateId = crac.getPreventiveState().getId();
+        crac.getNetworkActions().forEach(na -> {
+            NetworkActionResult networkActionResult = na.getExtension(NetworkActionResultExtension.class).getVariant(cracVariantId);
```
Comment (alexandrem-rte) [inline data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java:35]: What if na.getExtension(NetworkActionResultExtension.class) is null ? Maybe we should add a check here.

File: data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            if (networkActionResult != null) {
+                if (networkActionResult.isActivated(preventiveStateId)) {
+                    na.apply(network);
+                }
+            } else {
+                LOGGER.error(String.format("Could not find results for variant %s on network action %s", cracVariantId, na.getId()));
+            }
+        });
+        crac.getRangeActions().forEach(ra -> {
+            RangeActionResult rangeActionResult = ra.getExtension(RangeActionResultExtension.class).getVariant(cracVariantId);
```
Comment (alexandrem-rte) [inline data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java:45]: Same remark for ra.getExtension(RangeActionResultExtension.class)

File: data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java (hunk 1 above)
Comment (murgeyseb) [reply data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java]: Done

File: data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java (hunk 2 above)
Comment (murgeyseb) [reply data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/CracResultUtil.java]: Done

**Review by alexandrem-rte — APPROVED** [review]

**Pull request merged**